jdaviz/configs/default/plugins/virtual_observatory/vo_plugin.py

```diff
@@ -0,0 +1,419 @@
+
+        # If the user panned but tracking not enabled, don't recenter
+        # NOTE: float event typecheck assumes the only "float" type event triggering this method
+        #   is "zoom_center_x" and "zoom_center_y"
+        #   Otherwise, need some other way to determine the event = user panning
+        if isinstance(event, float) and not self.coord_follow_viewer_pan:
+            # Thus, we're no longer centered
+            self.viewer_centered = False
+            return
+
+        # gets the current viewer
+        if self.viewer_selected:
+            viewer = self.viewer.selected_obj
+
+        # nothing happens in the case there is no image in the viewer
+        # additionally if the data does not have WCS
+        if (
+            viewer.state.reference_data is None
+            or viewer.state.reference_data.coords is None
+        ):
+            self.source = ""
+            return
+
+        # Obtain center point of the current image and convert into sky coordinates
+        x_center = (viewer.state.x_min + viewer.state.x_max) * 0.5
+        y_center = (viewer.state.y_min + viewer.state.y_max) * 0.5
+        skycoord_center = viewer.state.reference_data.coords.pixel_to_world(
+            x_center, y_center
+        )
+
+        # Extract SkyCoord values as strings for plugin display
+        ra_deg = skycoord_center.ra.deg
+        dec_deg = skycoord_center.dec.deg
+        frame = skycoord_center.frame.name.lower()
+
+        # Show center value in plugin
+        self.source = f"{ra_deg} {dec_deg}"
+        self.coordframe_selected = frame
+
+        self.viewer_centered = True
+
+    @observe("waveband_selected", "change")
+    def vue_query_registry_resources(self, _=None):
+        """
+        Query Virtual Observatory registry for all SIA services
+        that serve data in that waveband around the source.
+        Then update the dropdown accordingly.
+        """
+        # If waveband was changed to nothing, immediately quit
+        # Don't throw an error due to trigger by plugin init
+        if not self.waveband_selected:
+            return
+
+        # Can't filter by coverage if we don't have a source to filter on
+        if self.resource_filter_coverage and not self.source:
+            error_msg = (
+                "Source is required for registry querying when coverage filtering is enabled. "
+                + (
+                    "Please enter your coordinates above "
+                    if self.viewer == "Manual"
+                    else f"Load data into viewer {self.viewer} first before querying "
+                )
+                + "or disable coverage filtering."
+            )
+            self.hub.broadcast(SnackbarMessage(error_msg, sender=self, color="error"))
+            raise ValueError(error_msg)
+
+        # Clear existing resources list
+        self.resources = []
+        self.resource_selected = None
+        self.resources_loading = True  # Start loading bar
```

Comment:
this could probably make use of the `with_spinner` decorator so you don't need to set/unset manually.

Reply:
Cool! I'm so used to doing it manually, I didn't notice you made a mixin for it! Awesome!

Done in https://github.com/spacetelescope/jdaviz/pull/2872/commits/4993e9284a33081321013bd29e0670c309c99c2f